inc/class-core-author-block.php

```diff
@@ -0,0 +1,277 @@
+			// Count the bylines.
+			$bylines_count = count( $this::$bylines['profiles'] );
+
+			// A string variable that will be used for the final rendered content.
+			$additional_blocks = '';
+
+			// Loop through the total bylines count.
+			for ( $i = 0; $i < $bylines_count; $i++ ) {
+				// Filter the block and append it to our string variable.
+				$additional_blocks .= $this->filter_post_author_block( $block_content, $i );
+			}
+
+			// Replace the original block with the new blocks.
+			$block_content = $additional_blocks;
+
+			// Update our boolean to not render anymore.
+			$this::$render = false;
+		} else {
+			return '';
+		}
+
+		return $block_content;
+	}
+
+	/**
+	 * Filters the author block data with byline info
+	 *
+	 * @param string $block_content The block content.
+	 * @param int    $i Used as an array key.
+	 *
+	 * @return string The modified core author block.
+	 */
+	public function filter_post_author_block( string $block_content, int $i ): string {
+		// Setup some empty variables.
+		$byline_type      = '';
+		$profile_post     = '';
+		$text_byline      = '';
+		$new_author_block = '';
+		$profiles         = $this::$bylines['profiles'];
+
+		// Bail if the array item does not exist.
+		if ( empty( $profiles[ $i ] ) ) {
+			return $block_content;
+		}
+
+		// Check if the byline uses a Profile Post ID.
+		if ( $this->validate_byline_post( $profiles[ $i ] ) ) {
+			// Get the byline post.
+			$profile_post = get_post( $profiles[ $i ]['atts']['post_id'] );
+			$byline_type  = 'profile';
+		}
+
+		// Check if the byline uses a regular text.
+		if ( $this->validate_byline_text( $profiles[ $i ] ) ) {
+			// Get the byline text.
+			$text_byline = $profiles[ $i ]['atts']['text'];
+			$byline_type = 'text';
+		}
+
+		// Check if the profile is a text field.
+		if ( ! empty( $text_byline ) && 'text' === $byline_type ) {
+			// Get the new block.
+			$new_author_block = $this->replace_author_block_name( $block_content, $text_byline );
+		}
+
+		// Check if the profile is a WP_Post.
+		if ( 'profile' === $byline_type ) {
+			if ( is_null( $profile_post ) || is_string( $profile_post ) || ! is_a( $profile_post, 'WP_Post' ) ) {
+				// Return the original block if something is wrong with the profile post.
+				return $block_content;
+			}
+
+			// Get the new block.
+			$new_author_block = $this->replace_author_block_author( $block_content, $profile_post );
+		}
+
+		// Make sure the new block is valid.
+		if ( $new_author_block && is_string( $new_author_block ) ) {
+			$block_content = $new_author_block;
+		}
+
+		return $block_content;
+	}
+
+	/**
+	 * Replaces the author name, avatar, bio and link in the author block.
+	 *
+	 * @param string  $html Author block HTML.
+	 * @param WP_Post $profile_post Post object of the profile post type.
+	 *
+	 * @return bool|string Filtered author block.
+	 */
+	public function replace_author_block_author( string $html, WP_Post $profile_post ): bool|string {
```

Comment:
I think it'd be great to have unit tests for this. Just to confirm this is doing what we expect it to do years from now.

Reply:
Working on this now.

Reply:
There's a unit test for this ready to go.